### PR TITLE
Improve exported types (and tweak a few tests)

### PR DESCRIPTION
**Pull request by smartcontracts**
Relatively simple PR, just exporting some additional types for `smock`. Will kick of some cleanup to `smock` in future PRs.

### CODE REVIEW

File: src/smockit/smockit.ts [hunk 1]
```diff
@@ -1,19 +1,16 @@
...
+import { ContractInterface, ethers } from 'ethers'
 
 /* Imports: Internal */
-import { MockContract, MockReturnValue } from './types'
+import { MockContract, MockReturnValue, SmockOptions, SmockSpec } from './types'
 import { bindSmock } from './binding'
 import { toHexString, fromHexString, makeRandomAddress } from '../utils'
 
 export const smockit = async (
-  spec: ContractInterface | Contract | ContractFactory | any,
```
Comment (smartcontracts) [inline src/smockit/smockit.ts:11]: Moved these into their own file.

File: src/smockit/types.ts [hunk 2]
```diff
@@ -1,5 +1,17 @@
...
+
+export type SmockSpec =
+  | ContractInterface
+  | Contract
+  | ContractFactory
+  | string
+  | any
+
+export interface SmockOptions {
+  provider?: any // What's the right type for a generic ethers provider?
```
Comment (smartcontracts) [inline src/smockit/types.ts]: Could use some assistance here if anyone knows.

File: src/smockit/types.ts [hunk 3]
```diff
@@ -1,5 +1,17 @@
 /* Imports: External */
-import { Contract } from 'ethers'
+import { Contract, ContractFactory, ContractInterface } from 'ethers'
+
+export type SmockSpec =
+  | ContractInterface
+  | Contract
+  | ContractFactory
+  | string
+  | any
```
Comment (smartcontracts) [inline src/smockit/types.ts]: "any" here really needs to be the structure of the standard JSON abi. Not sure if that type is easily accessible anywhere?

File: src/smockit/types.ts (hunk 2 above)
Comment (tynes) [reply src/smockit/types.ts]: `AbstractProvider`

Comment (tynes) [reply src/smockit/types.ts]: See https://github.com/ethereum-optimism/optimism-monorepo/pull/415/files

Comment (smartcontracts) [reply src/smockit/types.ts]: Nice.

**Pull request merged**